ui.frontend/src/utils.js

```diff
@@ -227,8 +227,12 @@ export default class Utils {
     static registerClickHandler(parentElement, hook, handler) {
         const dataAttr = "[" + hook + "]";
         const buttons = parentElement.querySelectorAll(dataAttr);
-        buttons.forEach(
-            (button) => button.addEventListener('click', handler));
+        buttons.forEach((button) => {
+          if(!button.getAttribute('data-event-registered')){
```

Comment:
we should get rid of extra data attributes if they can be avoided to simplify public api surface.

Reply:
consider using `:scope` to query for direct descendent button elements.